lib/importer/factory/object_factory.rb

```diff
@@ -136,6 +81,85 @@ def create
       log_created(object)
     end
 
+    def update
+      raise "Object doesn't exist" unless object
+
+      %w[created issued notes].each do |prop|
+        clear_attribute!(object, prop)
+      end
+
+      object.attributes = update_attributes
+
+      # TODO: #attach_files needs to always run for ETD ingests, since
+      # it triggers {Proquest::Metadata#run} which updates the
+      # {AdminPolicy} of the ETD itself. Currently on ETD ingests,
+      # `files' is never empty, so for now this is OK.
+      if files.empty?
+        $stderr.puts "No files provided for #{object.id}"
+      else
+        attach_files(object, files)
+        render_thumbnails(object)
+      end
+
+      run_callbacks(:save) do
+        object.save!
+      end
+      log_updated(object)
+    end
+
+    # Overridden in classes that inherit from ObjectFactory
+    #
+    # @param [Hash] object
+    # @param [String, Hash] files Either the path to the file or a
+    #     hash with file metadata
+    def attach_files(_object, _files)
+      raise NotImplementedError,
+            "#attach_files is not defined for #{self.class}"
+    end
+
+    def render_thumbnails(object)
```

Comment:
@supritasrinivas I organized ObjectFactory so the diff is a little confusing; here's the new method.